### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by kyrylomanko**
[DEMO LINK](https://kyrylomanko.github.io/react_redux-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,64 @@
...
+    return todos
+      .filter(todo => {
+        switch (filter.status) {
+          case Status.Active:
+            return !todo.completed;
+          case Status.Completed:
+            return todo.completed;
+          default:
+            return true;
+        }
```
Comment (volodymyr-soltys97) [inline src/App.tsx:35]:
```suggestion
        switch (filter.status) {
          case Status.Active:
            return !todo.completed;
            
          case Status.Completed:
            return todo.completed;
            
          default:
            return true;
        }
```

File: src/components/TodoModal/TodoModal.tsx [hunk 2]
```diff
@@ -1,42 +1,69 @@
...
+  const dispatch = useAppDispatch();
+
+  const handleOnClose = () => {
+    dispatch(currentTodoSlice.actions.removeTodo(null));
+    setUser(null);
+  };
+
+  useEffect(() => {
+    getUser(selectedTodo.userId)
+      .then(data => setUser(data))
```
Comment (volodymyr-soltys97) [inline src/components/TodoModal/TodoModal.tsx:21]:
```suggestion
      .then(setUser)
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
You need to fix the search logic 
image
Working example
image

Also you need to fix all failed tests feel free in the fe_chat if you need help

File: src/components/TodoModal/TodoModal.tsx (hunk 2 above)
Comment (kyrylomanko) [reply src/components/TodoModal/TodoModal.tsx]: Idk why the tests fail, when i do it manually everything's fine

**Pull request closed without merging**